src/taskpane/server.js

```diff
@@ -1789,6 +1985,24 @@ function getHierarchicalParentId(indent, previousEntries) {
   return -1;
 }
 
+// returns the correct parentId for the relevant group position by looping back through the list of entries
+// returns -1 if no match found
+// @param: entriesForExport - the whole list of artifacts in the spreadsheet
+// @param: i - target artifact position to look for
+// @param: artifactId - artifact type
+function getAssociationParentId(entriesForExport, i, artifactId) {
+  // if there is no indent/ set to initial indent we return out immediately 
+  if (artifactId == 2 || artifactId == 7) {
+
+    for (i; i > 0; i--) {
+
+      if (entriesForExport[i - 1].hasOwnProperty('TestCaseId')) {
+        return entriesForExport[i - 1].TestCaseId;

```

Comment:
you should not be hard coding a specific field here - this is a generic function, but looks like it is actually only about test cases. This is very confusing. I suggest you make the function be more generic and handle all cases like test cases, not just test cases

Reply:
Changes done - the code now uses generic objects and enumerations